packages/xpetra/src/MultiVector/Xpetra_EpetraMultiVector.hpp

```diff
@@ -479,7 +479,9 @@ namespace Xpetra {
       // scale().  Deal with this by scaling one column at a time.
       const size_t numVecs = this->getNumVectors ();
       for (size_t j = 0; j < numVecs; ++j) {
-        vec_->Scale (alpha[j]);
+        Teuchos::ArrayRCP< Scalar > vec = getDataNonConst(j);
```

Comment:
What was wrong with the call to `Scale`?

Reply:
Epetra's `Scale` scales all columns of the MultiVector by the same value. What we want is that the j-th column gets scaled by `alpha[j]`. So the previous implementation effectively multiplied the underlying `Epetra_MultiVector` by `alpha[0] * alpha[1] * alpha[2] * ..`

Reply:
Why not just get the Vector corresponding to column j, and call Scale on that with `alpha[j]`?  Code at the Xpetra level should be as abstract as possible.

Reply:
Because I didn't find the correct call in the documentation :-/
Thanks for pointing that out!